### PR TITLE
Don't build std for *-uefi targets

### PR DESCRIPTION
**Pull request by nicholasbishop**
https://github.com/rust-lang/rust/issues/97322

### CODE REVIEW

File: src/bootstrap/config.rs [hunk 1]
```diff
@@ -411,7 +411,11 @@ pub struct Target {
 impl Target {
     pub fn from_triple(triple: &str) -> Self {
         let mut target: Self = Default::default();
-        if triple.contains("-none") || triple.contains("nvptx") || triple.contains("switch") {
+        if triple.contains("-none")
+            || triple.contains("nvptx")
+            || triple.contains("switch")
+            || triple.contains("-uefi")
```
Comment (jyn514) [inline src/bootstrap/config.rs:417]: I don't recognize these other targets. "switch" is the Nintendo switch? What's nvptx? How did you determine these don't have std available?

Comment (nicholasbishop) [reply src/bootstrap/config.rs:417]: Those were already there, rustfmt just put them on different lines due to the single line getting too long.

Comment (jyn514) [reply src/bootstrap/config.rs:417]: Ugh, rustfmt strikes again. Thanks.

**Comment by jyn514** [comment]
@bors r+ rollup

We should probably find a better long term solution, like the issue mentioned ... but this is fine for now.

**Pull request merged**